### PR TITLE
chore(Java) : add circular singly linked list

### PR DESCRIPTION
**Pull request by riteshyadav27**
Singly Circular LinkedList with functionality of Add, Add in front, Display, Reverse , Search element in LinkedList

### CODE REVIEW

**Review by Parsh092 — APPROVED** [review]

**Review by Parsh092 — DISMISSED** [review]

**Comment by riteshyadav27** [comment]
> Thanks for opening this pull request! Please check out our contributing guidelines.
> 

I checked that and all requirements were done.

**Comment by riteshyadav27** [comment]
I have added BFS and DFS code for graph traversal in java.

**Review by mohitchakraverty — CHANGES_REQUESTED** [review]
Please also update the respective README.md file
In your case, it will be algorithms/Java/README.md

**Comment by riteshyadav27** [comment]
I updated the README.md file for circular singly linked list in Linked List Section , BFS and DFS in Graph Section .

**Review by mohitchakraverty — CHANGES_REQUESTED** [review]
Please add more documentation to your code that helps users understand better.

**Comment by riteshyadav27** [comment]
I have added the Output of all programs with their proper input .

**Review by mohitchakraverty — CHANGES_REQUESTED** [review]
- Please make different PRs for different algorithm
- In naming the file, replace underscore with dashes
- Change the name circularll to circular-singly-linkedlist
- Also add comments to your code explaining what or how your code is working

**Comment by riteshyadav27** [comment]
Updated Readme.md file, changed file name, explanation added for circular-singly-linkedlist having reverse(), addfront(), search() functions with Output

**Comment by riteshyadav27** [comment]
@mohitchakraverty will you please add hacktoberfest 2022 label 🙌

**Comment by riteshyadav27** [comment]
@mohitchakraverty I didn't understand why codespell is failed . Will you please help me what is it and how it works .

File: algorithms/Java/linked-lists/circular-singly-linkedlist.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+The Linkedlist is 5 8 7 1 2 
+The Linkedlist is 9 5 8 7 1 2 
+Target is found
+Target Not Found
+The Linkedlist is 2 1 7 8 5 9 
+
+*/
+
+// Here intial linkedlist was 9 5 8 7 1 2 and after using reverse function the linkedlist become 2 1 7 8 5 9
```
Comment (mohitchakraverty) [inline algorithms/Java/linked-lists/circular-singly-linkedlist.java:119]:
```suggestion
// Here initial linkedlist was 9 5 8 7 1 2 and after using reverse function the linkedlist become 2 1 7 8 5 9
```

**Review by mohitchakraverty — CHANGES_REQUESTED** [review]
Also fix the spelling error in DSA/algorithms/CPlusPlus/Arrays/max-subarray-sum.cpp
At line line 23 : Approach

Then you are good to go and your code will pass all the test.

**Comment by riteshyadav27** [comment]
@mohitchakraverty Done. I updated both the files.

**Review by mohitchakraverty — APPROVED** [review]

**Comment by mohitchakraverty** [comment]
Great work! Thanks

**Pull request merged**